runtime/util_core/j9shchelp.c

```diff
@@ -357,9 +387,24 @@ isCacheFileName(J9PortLibrary* portlib, const char* nameToTest, uintptr_t expect
 		}
 	}
 	nameToTestLen = strlen(nameToTest);
-	if ((nameToTest[nameToTestLen-3] != 'G') && (nameToTest[nameToTestLen-4] != '_')) {
-		return 0;
+	/**
+	 * For old cache file with no layer number like C290M4F1A64P_CC1_G37, 'G' is at nameToTestLen-3.
+	 * For cache file with layer number like C290M4F1A64P_CC1_G39L01, 'G' is at nameToTestLen-6.
+	 */
+	if (getLayerFromName(nameToTest) == J9SH_LAYER_NUM_UNSET) {
```

Comment:
This should check that `nameToTestLen` is large enough.

Reply:
Check added